backends/imgui_impl_dx9.cpp

```diff
@@ -70,6 +75,435 @@ static ImGui_ImplDX9_Data* ImGui_ImplDX9_GetBackendData()
     return ImGui::GetCurrentContext() ? (ImGui_ImplDX9_Data*)ImGui::GetIO().BackendRendererUserData : NULL;
 }
 
+// Shader pipeline
+static bool ImGui_ImplDX9_Shader_CreatePipeline()
+{
+    ImGui_ImplDX9_Data* bd = ImGui_ImplDX9_GetBackendData();
+    
+    D3DCAPS9 caps; ZeroMemory(&caps, sizeof(D3DCAPS9));
+    if (D3D_OK != bd->pd3dDevice->GetDeviceCaps(&caps))
+        return false;
+    if (caps.VertexShaderVersion < D3DVS_VERSION(2,0) || caps.PixelShaderVersion < D3DPS_VERSION(2, 0))
+        return false;
+    
+    // Input layout of default ImDrawVert
+    static const D3DVERTEXELEMENT9 InputLayout[4] = {
+        { 0,  0, D3DDECLTYPE_FLOAT2  , D3DDECLMETHOD_DEFAULT, D3DDECLUSAGE_POSITION, 0 },
+        { 0,  8, D3DDECLTYPE_FLOAT2  , D3DDECLMETHOD_DEFAULT, D3DDECLUSAGE_TEXCOORD, 0 },
+        { 0, 16, D3DDECLTYPE_D3DCOLOR, D3DDECLMETHOD_DEFAULT, D3DDECLUSAGE_COLOR   , 0 },
+        D3DDECL_END(),
+    };
+    if (D3D_OK != bd->pd3dDevice->CreateVertexDeclaration(InputLayout, &bd->pInputLayout))
+        return false;
+    
+    // Shaders
+    #define CODE(b1, b2, b3, b4) 0x##b1, 0x##b2, 0x##b3, 0x##b4
+    static const BYTE VertexShaderByteCode[] = {
+        // HLSL source code
+        /*
+            float4x4 mvp : register(c0);
+            
```

Comment:
Suggest compacting the source code a little, remove empty lines, compact structures into one-liner.
The reasoning is to make this "optional" feature of DX9 backend not takes too much visibility in the source file, making it more likely to be mergeable.

Reply:
ok, I deleted redundant empty lines